growth/too/models.py

```diff
@@ -727,6 +728,169 @@ def flat(self):
+
+        ipix = np.asarray(list(self.observation_ipix(filt, start_time, end_time)))
+        if len(ipix) > 0:
+            return self.flat_2d[ipix].sum()
+        else:
+            return 0.0
+
+    def get_observations(self, telescope, filt, start_time, end_time):
+
+        # localization = Localization.query.filter_by(
+        #     dateobs=self.dateobs,
+        #     localization_name=self.localization_name).one()
+        prob = self.flat_2d
+
+        # fields = Field.query.filter_by(telescope=telescope).all()
+        # field_prob_1, field_ids_1 = [], []
+        # field_prob_2, field_ids_2 = [], []
+        # for field in fields:
+        #     if field.field_id > 800 and telescope == 'ZTF':
+        #         field_ids_2.append(field.field_id)
+        #         field_prob_2.append(np.sum(prob[field.ipix]))
+        #     else:
+        #         field_ids_1.append(field.field_id)
+        #         field_prob_1.append(np.sum(prob[field.ipix]))
+
+        fields = Field.query.filter_by(telescope=telescope)
+
+        # calculate the overlap between the field ipix and the ipix corresponding to the prob in the 90 %
+        
+        # a list containing the credible level corresponding to each ipix
+        levels = self.credible_levels_2d()
+        idx = levels <= 0.90
+        # all of the ipix corresponding to the 90% region
+        localization_ipix = range(len(prob))[idx]
+        # select the fields that overlap with the ipix array; select fields where ipix && 90% ipix; && indicates overlap between tables
+
+        for field in fields:
+            overlap = field.overlap(ipix=localization_ipix)
+            if overlap == True: field_ids.append(field_id)
+
+
+        # if field.field_id > 800 and self.telescope == 'ZTF': 
+
+        #     field_ids_1, field_ids_2 = np.array(field_ids_1), np.array(field_ids_2)
+        #     field_prob_1 = np.array(field_prob_1)
+        #     field_prob_2 = np.array(field_prob_2)
+
+        #     field_ids_1 = field_ids_1[np.argsort(field_prob_1)[::-1]]
+        #     field_prob_1 = np.sort(field_prob_1)[::-1]
+
+        #     field_ids_2 = field_ids_2[np.argsort(field_prob_2)[::-1]]
+        #     field_prob_2 = np.sort(field_prob_2)[::-1]
+        #     cumsum = np.cumsum(field_prob_2)
+        #     idx2 = np.argmin(np.abs(cumsum-0.90))
+        #     field_ids_2 = field_ids_2[:idx2]
+        #     field_prob_2 = field_prob_2[:idx2]
+
+        #     cumsum = np.cumsum(field_prob_1)
+        #     idx2 = np.argmin(np.abs(cumsum-0.90))
+        #     field_ids_1 = field_ids_1[:idx2]
+        #     field_prob_1 = field_prob_1[:idx2]
+
+        #     field_ids = np.hstack((field_ids_1,field_ids_2))
+        
+        # else: field_ids = field_ids_1
+
+        telescope = Telescope.query.filter_by(telescope=self.telescope).one()
+        filts = list(telescope.filters)
+        # filts.append('a')
+
+        bands = {'g': 1, 'r': 2, 'i': 3, 'z': 4, 'J': 5, 'U': 6}
```

Comment:
Is this hardcoded dictionary really not part of the data model already?

Reply:
unfortunately, no. it seems to be hardcoded in three places, including this one. Where would you recommend adding it into the data model?

Reply:
Yes, but later, not in this pull request.